launch:
# args that can be set from the command line or a default will be used
- arg:
    name: "fps"
    default: "30"
- arg:
    name: "frame_width"
    default: "640"
- arg:
    name: "frame_height"
    default: "480"

- node:
    pkg: "depth_projector"
    exec: "camera_pub"
    name: "camera_pub"
    namespace: "depth_projector"
    param:
    - 
        name: "fps"
        value: "$(var fps)"
    -
        name: "frame_width"
        value: "$(var frame_width)"
    -
        name: "frame_height"
        value: "$(var frame_height)"

- node:
    pkg: "depth_projector"
    exec: "depth_publisher"
    name: "depth_publisher"
    namespace: "depth_projector"
    param:
    -
        name: "frame_width"
        value: "$(var frame_width)"
    -
        name: "frame_height"
        value: "$(var frame_height)"

- node:
    pkg: "depth_projector"
    exec: "projector"
    name: "projector"
    namespace: "depth_projector"
